### DISCUSSION TITLE
Expose allocated memory from C++ to python in trampoline class

### DISCUSSION THREAD

**Discussion by ThoreWietzke** [wjakob/nanobind#331] [General]
This thread is the outcom from issue https://github.com/wjakob/nanobind/issues/330.

I have the following function, which allocates memory and calls a member function of a class which modifies the data in the allocated Memory:
```c++
void foo(Base* base)
{
    double* data = allocateMemory();
    Eigen::Map<Eigen::VectorXd> mappedData (data, n);
    base->func(mappedData);
    // do something with data
}
```
The base class looks like
```c++
class Base
{
    public:
        ~Base() {};
        virtual void func(Eigen::Ref<Eigen::VectorXd> in) = 0;
};
```
with a trampoline to Python
```c++
class PyBase : public Base 
{
    NB_TRAMPOLINE(Base, 1);
    public:

        double func(Eigen::Ref<Eigen::VectorXd> in) override
        {
            NB_OVERRIDE_PURE(
                func, /*Name of function*/
                in /*arguments*/   
            );
        }
};
```

I want to expose the allocated memory inside `foo()` in the corresponding Python implementation of `func()`. In pybind11 this worked.
Is this the way to go in nanobind? Are there other ways to expose allocated memory to Python?

**Comment by wjakob** [comment]
This issue is resolved given that #334 is merged.
